File: PersonalAssistantUI/src/app/models/message/message.ts
import { User } from "..";
import { Room } from "..";

export class Message {
    public id?: number;
    public senderId: number;
    public sender?: User;
    public roomId: number;
    public room?: Room;
    public receiverType: string;
    public type?: string; // TEXT
    public category?: string;
    public sentAt?: Date;
    public status?: string;
    public editedAt?: Date;
    public editedBy?: string;
    public deletedAt?: Date;
    public deletedBy?: number;
    public text: string;
    public parentMessageId?: number;

    constructor(senderId: number, roomId: number, text: string, type: string, receiverType: string) {
        this.senderId = senderId;
        this.roomId = roomId;
        this.text = text;
        this.receiverType = receiverType;
        this.type = type;
        this.sentAt = new Date(Date.now());
    }
}

File: PersonalAssistant.DataAccess/Entities/Message.cs
using System;
using PersonalAssistant.DataAccess.Identity;

namespace PersonalAssistant.DataAccess.Entities
{
    public class Message
    {
        public int Id { get; set; }
        public int SenderId { get; set; }
        public User Sender { get; set; }
        public int? RoomId { get; set; }
        public Room Room { get; set; }
        public string Type { get; set; }
        public string Category { get; set; } //CometChat.CATEGORY_MESSAGE,//CometChat.MESSAGE_TYPE.CUSTOM,//CometChat.CATEGORY_ACTION,//CometChat.CATEGORY_CALL,
        public DateTime? SentAt { get; set; }
        public DateTime? EditedAt { get; set; }
        public int EditedBy { get; set; }
        public DateTime? DeletedAt { get; set; }
        public int DeletedBy { get; set; }
        public string Text { get; set; } // TODO separate class
        public int? ParentMessageId { get; set; }
    }
}
